src/traces/scattergl/index.js

```diff
@@ -13,10 +13,12 @@ var ScatterGl = {};
 ScatterGl.attributes = require('./attributes');
 ScatterGl.supplyDefaults = require('./defaults');
 ScatterGl.colorbar = require('../scatter/colorbar');
+ScatterGl.hoverPoints = require('../scatter/hover');
 
 // reuse the Scatter3D 'dummy' calc step so that legends know what to do
-ScatterGl.calc = require('../scatter3d/calc');
+ScatterGl.calc = require('../scatter/calc');
```

Comment:
@etpinard we can effortlessly get the `mouseContainer` back, it is just a matter of 2 commented LOC.
I will try to optimize _snap-points-2d_ in _gl-scatter2d_ update, I feel optimistic about that, it should add decent selection performance, because now it regenerates hierarchical cache every draw frame for a millions of points.
As for modifying [cartesian/select](https://github.com/plotly/plotly.js/blob/master/src/plots/cartesian/select.js) that implies implementing part of [gl2d/camera](https://github.com/plotly/plotly.js/blob/78b82b85206bde93c7117706278ac698b480139d/src/plots/gl2d/camera.js) in it, but that seems to be the only choice if we want to keep gl picker.